docs/pipelines.md

```diff
@@ -75,19 +83,26 @@ A `Pipeline` definition supports the following fields:
 - Optional:
   - [`resources`](#specifying-resources) - **alpha only** Specifies
     [`PipelineResources`](resources.md) needed or created by the `Tasks` comprising the `Pipeline`.
+  - [`params`](#specifying-parameters) - Specifies the `Parameters` that the `Pipeline` requires.
+  - [`workspaces`](#specifying-workspaces) - Specifies a set of Workspaces that the `Pipeline` requires.
   - [`tasks`](#adding-tasks-to-the-pipeline):
-      - `resources.inputs` / `resource.outputs`
-        - [`from`](#using-the-from-parameter) - Indicates the data for a [`PipelineResource`](resources.md)
+      - [`resources`](#specifying-resources-in-pipelinetasks) - Specifies the [`PipelineResource`](resources.md) that
+        a `Task` requires.
+        - [`from`](#using-the-from-field) - Indicates the data for a [`PipelineResource`](resources.md)
           originates from the output of a previous `Task`.
-      - [`runAfter`](#using-the-runafter-parameter) - Indicates that a `Task`
-        should execute after one or more other `Tasks` without output linking.
-      - [`retries`](#using-the-retries-parameter) - Specifies the number of times to retry the
-        execution of a `Task` after a failure. Does not apply to execution cancellations.
+      - [`runAfter`](#using-the-runafter-field) - Indicates that a `Task` should execute after one or more other 
+        `Tasks` without output linking.
+      - [`retries`](#using-the-retries-field) - Specifies the number of times to retry the execution of a `Task` after
+        a failure. Does not apply to execution cancellations.
       - [`conditions`](#guard-task-execution-using-conditions) - Specifies `Conditions` that only allow a `Task`
         to execute if they successfully evaluate.
+      - [`when`](#guard-finally-task-execution-using-when-expressions) - Specifies `when` expressions that guard
+        the execution of a `Task`; allow execution only when all `when` expressions evaluate to true.
       - [`timeout`](#configuring-the-failure-timeout) - Specifies the timeout before a `Task` fails.
-  - [`results`](#configuring-execution-results-at-the-pipeline-level) - Specifies the location to which
-    the `Pipeline` emits its execution results.
+      - [`params`](#specifying-parameters-in-pipelinetasks) - Specifies the `Parameters` that a `Task` requires.
+      - [`workspaces`](#specifying-workspaces-in-pipelinetasks) - Specifies the `Workspaces` that a `Task` requires.
+  - [`results`](#emitting-results-from-a-pipeline) - Specifies the location to which the `Pipeline` emits its execution
+    results.
   - [`description`](#adding-a-description) - Holds an informative description of the `Pipeline` object.
   - [`finally`](#adding-finally-to-the-pipeline) - Specifies one or more `Tasks`
```

Comment:
are we listing a comprehensive list of supported fields? shall we update `finally` to include all the supported fields? (can be a followup PR)